packages/util/src/assignment.ts

```diff
@@ -165,18 +165,19 @@ export function meetupLocation(meetupIndex: MeetupIndexType, locations: Vec<Loca
 
 /**
  * Get the meetup time for a given location.
- *
- * @param location
- * @param attesting_start
- * @param one_day
  */
-export function meetupTime(location: Location, attesting_start: Moment, one_day: Moment): Moment {
+export function meetupTime(location: Location, attesting_start: Moment, one_day: Moment, offset: MeetupTimeOffsetType): Moment {
     const registry = location.registry;
 
     const per_degree = one_day.toNumber() / 360;
-    const lon_time = parseDegree(location.lon) * per_degree;
 
-    let result = Math.round(attesting_start.toNumber() + one_day.toNumber() / 2 - lon_time);
+    // The meetups start at high sun at 180 degrees and during one day the meetup locations travel
+    // along the globe until the very last meetup happens at high sun at -180 degrees.
+    // So we scale the range 180...-180 to 0...360
+    const lon = Math.abs(parseDegree(location.lon) - 180);
+    const lon_time = lon * per_degree;
+
+    let result = Math.round(attesting_start.toNumber() + lon_time + offset.toNumber());
```

Comment:
if we have an offset, that means that at 179° or -179° degrees there is a risk for a date change. But as we work with an epoch timestamp that should not lead to issues. But what could happen is that at -180° and a negative offset, the meetup time will be before the attesting phase even starts. This is not necessarily a problem but attestations will only be able to be registered after attesting phase starts

Reply:
Yes you are right not very intuitive, but it is technically not a problem.